subsys/bluetooth/controller/ll_sw/ull_adv_iso.c

```diff
@@ -10,28 +10,396 @@
+	adv_iso->evt.ticks_xtal_to_start =
+		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_XTAL_US);
+	adv_iso->evt.ticks_preempt_to_start =
+		HAL_TICKER_US_TO_TICKS(EVENT_OVERHEAD_PREEMPT_MIN_US);
+	adv_iso->evt.ticks_slot = HAL_TICKER_US_TO_TICKS(slot_us);
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_LOW_LAT)) {
+		ticks_slot_overhead = MAX(adv_iso->evt.ticks_active_to_start,
+					  adv_iso->evt.ticks_xtal_to_start);
+	} else {
+		ticks_slot_overhead = 0;
+	}
+
+	/* TODO: Calculate ISO interval */
+	/* iso_interval shall be at least SDU interval,
+	 * or integer multiple of SDU interval for unframed PDUs
+	 */
+	iso_interval_us = adv_iso->sdu_interval;
+
+	ret_cb = TICKER_STATUS_BUSY;
+	ret = ticker_start(TICKER_INSTANCE_ID_CTLR, TICKER_USER_ID_ULL_HIGH,
+			   (TICKER_ID_ADV_ISO_BASE + adv_iso->bis_handle),
+			   ticks_anchor, 0,
+			   HAL_TICKER_US_TO_TICKS(iso_interval_us),
+			   HAL_TICKER_REMAINDER(iso_interval_us),
+			   TICKER_NULL_LAZY,
+			   (ll_adv_iso->evt.ticks_slot + ticks_slot_overhead),
+			   ticker_cb, ll_adv_iso,
+			   ull_ticker_status_give, (void *)&ret_cb);
+	ret = ull_ticker_status_take(ret, &ret_cb);
+
+	return 0;
+}
+
+static inline struct ll_adv_iso *ull_adv_iso_get(uint8_t handle)
+{
+	if (handle >= CONFIG_BT_CTLR_ADV_SET) {
+		return NULL;
+	}
+
+	return &ll_adv_iso[handle];
+}
 
 uint8_t ll_big_create(uint8_t big_handle, uint8_t adv_handle, uint8_t num_bis,
 		      uint32_t sdu_interval, uint16_t max_sdu,
 		      uint16_t max_latency, uint8_t rtn, uint8_t phy,
 		      uint8_t packing, uint8_t framing, uint8_t encryption,
-		      uint8_t *bcode)
+		      uint8_t *bcode, void **rx)
 {
-	/* TODO: Implement */
-	ARG_UNUSED(big_handle);
-	ARG_UNUSED(adv_handle);
-	ARG_UNUSED(num_bis);
-	ARG_UNUSED(sdu_interval);
-	ARG_UNUSED(max_sdu);
-	ARG_UNUSED(max_latency);
-	ARG_UNUSED(rtn);
-	ARG_UNUSED(phy);
-	ARG_UNUSED(packing);
-	ARG_UNUSED(framing);
-	ARG_UNUSED(encryption);
-	ARG_UNUSED(bcode[16]);
+	struct ll_adv_iso *adv_iso;
+	struct ll_adv_set *adv;
+	struct node_rx_pdu *node_rx;
+	struct pdu_bis *pdu_bis;
+	struct node_tx *tx;
 
-	return BT_HCI_ERR_CMD_DISALLOWED;
+	adv_iso = ull_adv_iso_get(big_handle);
+
+	if (!adv_iso) {
+		return BT_HCI_ERR_CMD_DISALLOWED;
+	}
+
+	adv = ull_adv_is_created_get(adv_handle);
+
+	if (!adv) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	} else if (!adv->lll.sync) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	} else if (adv->lll.sync->adv_iso) {
+		return BT_HCI_ERR_UNKNOWN_ADV_IDENTIFIER;
+	}
+
+	if (IS_ENABLED(CONFIG_BT_CTLR_PARAM_CHECK)) {
+		if (num_bis == 0 || num_bis > 0x1F) {
+			return BT_HCI_ERR_INVALID_PARAM;
+		}
+
+		if (sdu_interval < 0x000100 || sdu_interval > 0x0FFFFF) {
+			return BT_HCI_ERR_INVALID_PARAM;
+		}
+
+		if (max_sdu < 0x0001 || max_sdu > 0x0FFF) {
+			return BT_HCI_ERR_INVALID_PARAM;
+		}
+
+		if (max_latency > 0x0FA0) {
+			return BT_HCI_ERR_INVALID_PARAM;
+		}
+
+		if (rtn > 0x0F) {
+			return BT_HCI_ERR_INVALID_PARAM;
+		}
+
+		if (phy > 3) {
```

Comment:
phy bit 2 would be coded PHY, so values > 3 would be legal. Consider using defines here (cf. defines `BT_HCI_LE_EXT_SCAN_PHY_...`)